src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,7 +1625,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                if (!request.isSilent() && callbackHandle.callback != null && activity != null) {
+                    return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                } else {
+                    callbackHandle.onError(
+                            new AuthenticationException(ADALError.AUTH_FAILED_SERVER_ERROR, request.getLogInfo() + errLogInfo));
```

Comment:
Ok, it makes sense to be backward compat. But we'd like to distinguish between AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED from aquireTokenProvider and at this place. So let's extend Exception class.